website/docs/labs/capsule-exec.md

```diff
@@ -0,0 +1,162 @@
+---
+id: capsule-exec
+title: Exec syscall in Capsule
+---
+
+## Introduction
+
+[Exec](https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0034-vm-syscalls-2/0034-vm-syscalls-2.md#exec) is a new syscall provided by [ckb2021](https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0037-ckb2021/0037-ckb2021.md). To understand what exec syscall does, we recommend reading [this article](https://man7.org/linux/man-pages/man3/exec.3.html) first. In short: Exec runs an executable file from specified cell data in the context of an already existing VM, replacing the previous executable. The used cycles do not change, but the code, registers, and memory of the VM are replaced by those of the new program, meaning **control flow will never return to the main script**.
+
+You can imagine exec as a router. When some conditions are met, the main script will completely hand over control to a certain sub-script.
+
+```text
+              ┌--> if State1 --> Exec(Sub-script1)
+Main script --+--> if State2 --> Exec(Sub-script2)
+              └--> if State3 --> Exec(Sub-script3)
+```
+
+Compared with [Dynamic libraries](https://docs.nervos.org/docs/labs/capsule-dynamic-loading-tutorial/), exec has the following significant advantages:
+
+- All sub-scripts are complete scripts. They can be used alone, or they can be called by exec.
+- Sub-scripts have a separate 4M memory space.
+
```

Comment:
also add disadvantages: 1. never return 2. hard to exchanged data between scripts